sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeCreator.scala

```diff
@@ -87,8 +87,18 @@ case class GpuCreateNamedStruct(children: Seq[Expression]) extends GpuExpression
     case Seq(name, value) => (name, value)
   }.toList.unzip
 
+  // Names will be serialized before Spark scheduling, and the returned type GpuScalar
+  // from GpuLiteral.columnarEval(null) can't be serializable, which causes
+  // `org.apache.spark.SparkException: Task not serializable` issue.
+  //
+  // And on the other hand, the calling for columnarEval(null) in the driver side is
+  // dangerous for GpuExpressions, we'll have to pull it apart manually.
   private lazy val names = nameExprs.map {
-    case g: GpuExpression => g.columnarEval(null)
+    case ge: GpuExpression =>
+      GpuExpressionsUtils.extractGpuLit(ge) match {
```

Comment:
nit:
```scala
GpuExpressionsUtils.extractGpuLit(ge).map(_.value)
   .getOrElse(throw new IllegalStateException(s"Unexpected GPU expression $ge"))
```

Reply:
Thx Gera

Reply:
+1, but I would take it a step further because `extractGpuLit` takes an `Expression` as input not a `GpuExpression` and we should always get `GpuExpression`s in `nameExprs` or we have other problems.

```
private lazy val names = nameExprs.map { exp =>
  GpuExpressionsUtils.extractGpuLit(exp)
    .getOrElse(throw new IllegalStateException(s"Unexpected Expression $exp"))
}
```

Reply:
I don't think you understood. I don't want `extractGpuLit` to take a `GpuExpression`.  It is good like it is. I just don't think we need to check for the non-GpuExpression case when computing `names`.  It should never happen so it is OK to throw an exception if we see it., Which is why I said we could simplify the processing to just be.

```
private lazy val names = nameExprs.map { exp =>
  GpuExpressionsUtils.extractGpuLit(exp)
    .getOrElse(throw new IllegalStateException(s"Unexpected Expression $exp"))
}
```

Reply:
Oh, Yeah, You're right. Sorry for the misunderstanding.